_studio/shared/include/mfxvideo++int.h

```diff
@@ -299,6 +307,12 @@ class VideoPAK
     mfxStatus Close(void) = 0;
     virtual
     mfxTaskThreadingPolicy GetThreadingPolicy(void) {return MFX_TASK_THREADING_DEFAULT;}
+    virtual
+    mfxStatus GetThreadNum(mfxU32& threadNum) {
+        // some mediasdk components require 2 threads as a minimum
+        threadNum = std::max(std::thread::hardware_concurrency(), 2u);
```

Comment:
I believe resulting thread number should depend on threading policy, otherwise we have an ambiguity - default thread policy [MFX_TASK_THREADING_DEFAULT] states that we doesn't support parallel task execution, all task is executed by thread #0 only and, at the same time we request at least two threads (or number of hardware available ones)